### PR TITLE
Explicit group updates messages

### PR DESCRIPTION
**Pull request by Brice-W**
This PR is a small refactor of the closed group updates messages handling.
It introduces more explicit messages around group updates (create / name change / members added & removed / leaving).
It also introduces a new class that will handle the generation of the different update messages (closed groups, disappearing messages, etc)

### CODE REVIEW

File: libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageData.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.session.libsignal.utilities.JsonUtil
+import org.session.libsignal.utilities.logging.Log
+import java.util.*
+
+// class used to save update messages details
+class UpdateMessageData () {
+
+    var type: SignalServiceGroup.Type = SignalServiceGroup.Type.UNKNOWN
+    var groupName: String? = null
+    var updatedMembers: Collection<String> = Collections.emptyList()
```
Comment (nielsandriesse) [inline libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageData.kt:14]: These 3 lines seem like a prime candidate for a sealed class!

File: libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+object UpdateMessageBuilder {
+
+    fun buildGroupUpdateMessage(context: Context, updateData: UpdateMessageData, sender: String? = null, isOutgoing: Boolean = false): String {
+        val updateType = updateData.type
+        var message: String = ""
+        if (!isOutgoing && sender == null) return message
+        val senderName: String? = if (!isOutgoing) {
+            MessagingConfiguration.shared.storage.getDisplayNameForRecipient(sender!!) ?: sender
+        } else { sender }
```
Comment (nielsandriesse) [inline libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageBuilder.kt:17]: Should we maybe make this say "You" if the message is outgoing?

File: app/src/main/java/org/thoughtcrime/securesms/loki/utilities/GroupDescription.kt [hunk 3]
```diff
@@ -13,6 +13,7 @@ import network.loki.messenger.R
 import org.session.libsignal.utilities.logging.Log
 import java.io.IOException
 
+//TODO that class isn't used anymore
```
Comment (nielsandriesse) [inline app/src/main/java/org/thoughtcrime/securesms/loki/utilities/GroupDescription.kt:16]: Can we delete this class?

File: app/src/main/java/org/thoughtcrime/securesms/database/model/ThreadRecord.java [hunk 4]
```diff
@@ -73,7 +75,7 @@ public ThreadRecord(@NonNull String body, @Nullable Uri snippetUri,
   @Override
   public SpannableString getDisplayBody(@NonNull Context context) {
     Recipient recipient = getRecipient();
-    if (isGroupUpdate()) {
+    if (isGroupUpdate() || isGroupUpdateMessage()) {
```
Comment (nielsandriesse) [inline app/src/main/java/org/thoughtcrime/securesms/database/model/ThreadRecord.java:78]: What's still using `isGroupUpdate` at this point?

**Review by nielsandriesse — COMMENTED** [review]
Nice work @Brice-W :)

File: app/src/main/java/org/thoughtcrime/securesms/database/model/ThreadRecord.java (hunk 4 above)
Comment (Brice-W) [reply app/src/main/java/org/thoughtcrime/securesms/database/model/ThreadRecord.java:78]: The group updates messages prior to the release of that update will still need that test. We can get rid of it eventually, but for now it better to keep it.

File: app/src/main/java/org/thoughtcrime/securesms/loki/utilities/GroupDescription.kt (hunk 3 above)
Comment (Brice-W) [reply app/src/main/java/org/thoughtcrime/securesms/loki/utilities/GroupDescription.kt:16]: Sure! All I needed is a confirmation to delete it :)

File: libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageBuilder.kt (hunk 2 above)
Comment (Brice-W) [reply libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageBuilder.kt:17]: Yeah, makes sense, I'll correct that

File: libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageData.kt (hunk 1 above)
Comment (Brice-W) [reply libsession/src/main/java/org/session/libsession/messaging/utilities/UpdateMessageData.kt]: Yeah gotcha, I forgot about sealed classes, but that's definitely going to improve that class.
I added it. It was a bit tricky around json serialisation, but I figured it out.

**Pull request merged**